modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/ast/MacroExpansionModulesModel.java

```diff
@@ -86,6 +93,33 @@ public ArtifactAst expand() {
     }
 
     if (hasMacroExpansionExtension) {
+      // The macro expansion of xml sdk components causes inner flow components to be generated as a child of another component;
+      // so when generating the minimal app during lazy init, those flow components are not registered as top level components
+      // causing certain validations to fail. This code looks for those flow components and adds them as top level.
+      Stream<ComponentAst> flowComponents =
+          applicationModel.recursiveStream()
+              .filter(comp -> comp.getComponentType() != null && comp.getComponentType().equals(FLOW));
+      List<ComponentAst> componentsToAdd =
+          flowComponents.filter(comp -> !applicationModel.topLevelComponents().contains(comp)).collect(Collectors.toList());
```

Comment:
`!applicationModel.topLevelComponents().contains(comp)` shouldn't be needed if you are only navigating the extension being macroexpanded

Reply:
Even if dealing with only 1 extension, there is the case that the extension has 2 or more operations pointing to the same flow, so while travelling the ast you can find the flow twice, but you only want to add it once to top level. So this check should remain imo.